File: src/io/littleworld/course/choices.java

/// if
if (x < 3)
  System.out.println("small");
#// if x is smaller then 3 print 'small'

/// if with block
if (x < 3) {
  System.out.println("small");
}
#// if x is smaller then 3 print 'small'

/// if
if (x < 3)
  System.out.println("small");
else
  System.out.println("big");
#// if x is smaller then 3 print 'small'
else print 'big'

/// if shortcut
int y  = x < 3 ? -1 : 1;
#// if x is smaller then 3  y becomes -1
else y becomes 1

/// types
String name = ""
switch(x) {
  case 1: case 2:
      name = "one or two"; break;
  case 3: name = "three"; break;
  default: name = "?"
}
#// a jump table

File: src/io/littleworld/course/choices.kt

/// if
if (x < 3)
  println("small")
#// if x is smaller then 3 print 'small'

/// if
if (x < 3)
  println("small")
else
  println("big")
#// if x is smaller then 3 print 'small'
else print 'big'


/// short cut
int y  = if (x < 3) -1 else 1;
#// if x is smaller then 3  y becomes -1
else y becomes 1

/// when
var name = when(x) {
  1, 2 -> "one or two"
  3 -> "three"
  else -> "?"
}
#// pattern matching

/// while loop
var x = 0
while(x < 3){
  println(x)
  x++
}
#// as in Java

/// for loop
for (x in 0..3) {
  println(x)
}
#// prints a range [0..3]; with 3
